### PR TITLE
Update information for translating the interface with Crowdin

### LINKED ISSUE #16656: Add / update the documentation to translate NVDA's interface (nvda.po)
### Detailed description of the issue
New translators need guidance on how to translate the interface. Currently, there are many translators for new languages or abandoned ones, but they do not seem to go forward and some of them have explicitly asked for help on the translators mailing list.

For now, we have an outdated wiki as well as messages on the mailing list. For someone who has not followed the discussion during the transition, it is hard to know the current process to translate the interface as well as having tips on how to use Crodin.

I was about to write a wiki on how to translate the interface, when #16627 has been opened. Also I am not very comfortable with Crowdin.

### Why are the other templates not appropriate in this case?
It is documentation only.

### Have you asked for advice on how to report this issue via a community discussion? If so, please link to the discussion
No

### Steps to reproduce or illustrate the issue (if applicable)
N/A

### Expected outcome or behaviour (if applicable)
N/A

### Screenshots, logs, and other attachments (if applicable)
N/A

### System configuration or software environment (if applicable)
N/A

### Additional information (if applicable)
N/A

### PR DESCRIPTION
**Pull request by seanbudd**
### Link to issue number:
Closes #16656

### Summary of the issue:
NV Access is intending to move most of the wiki into the repository, so changes are tracked better and easier to propose.
The documentation for translating NVDA is very dated, particularly since the update of the translation system.
### Description of user facing changes

This wiki page was moved and updated in the project docs
- [Translating the interface](https://github.com/nvaccess/nvda/wiki/TranslatingTheInterface)

The following changes were made to update the contents:
- grammar and formatting fixes
- add reference to user guide commands for PoEdit
- remove section on translating using a text editor
- added information on:
  - on using Crowdin and proofreader status
  - plural forms
  - pgettext (context/tags)

### Future work
This wiki pages will be removed/deprecated when this PR is merged.
Moving and updating the rest of the translating wiki pages.

## Summary by CodeRabbit

- **Documentation**
  - Added a new guide for translating the NVDA interface using Crowdin.
  - Enhanced translation readme with improved formatting and updated references.

- **Chores**
  - Updated links in translation files to direct to the new Crowdin documentation.

### CODE REVIEW

File: projectDocs/translating/crowdin.md
Comment (CyrilleB79) [inline projectDocs/translating/crowdin.md:36]: The following information is missing and should be added in this paragraph:
* some word have plural forms; describe how to manage it, especially with poedit in which object navigation is required to reach the plural form
* some string have context information (when `pgettext` is used in the code)

File: projectDocs/translating/crowdin.md
Comment (CyrilleB79) [inline projectDocs/translating/crowdin.md:79]: With the real string in `nvda.po`, matching the example:
```suggestion
- `{color} on {backgroundColor}`: In this case, the word on should be translated, and the rest should be left alone.
```

or even with the word "on" formatted:
```suggestion
- `{color} on {backgroundColor}`: In this case, the word `on` should be translated, and the rest should be left alone.
```

File: projectDocs/translating/crowdin.md
Comment (CyrilleB79) [inline projectDocs/translating/crowdin.md:18]: If I am not mistaken, the job of the approver is not covered in this document. I think that you cannot synchronize poedit with Crowdin to approve strings; the only thing that you can do is to upload your .po file and check auto-approval when doing this operation. To be checked though.

But explaining how strings should be approved is required, especially for people who start translating a new language. Else, this document is useless for new translators.

**Review by CyrilleB79 — COMMENTED** [review]
As a general comment:

There is no paragraph to describe the usage of the web interface; does it mean that NVDA recommends poedit? poedit does not allow to see the unapproved translation proposals of other people.

My feeling is that there are 2 use cases to translate the interface for a specific language:
1. If there is only one translator of the interface for a specific language; in this case the usage of poedit is recommanded because it is easier. Note that to send a po file with approval; synchronization does not seem enough; you need to manually upload your file on Crowdin's web interface, checking the approval checkbox.
2. If there are many translators for the interface; in this case, the web interface, although less comfortable, cannot be avoided

File: projectDocs/translating/crowdin.md
Comment (seanbudd) [reply projectDocs/translating/crowdin.md:79]:
```suggestion
- `{color} on {backgroundColor}`: In this case, the word `on` should be translated.
The rest should be left alone or rearranged in order to suit the language.
```

File: projectDocs/translating/crowdin.md
Comment (seanbudd) [reply projectDocs/translating/crowdin.md:43]: I've added new sections here, let me know what you think.

File: projectDocs/translating/crowdin.md
Comment (seanbudd) [reply projectDocs/translating/crowdin.md]: I've added more information here

File: projectDocs/translating/crowdin.md
Comment (CyrilleB79) [inline projectDocs/translating/crowdin.md:40]: To be checked but, it seems to me, more precisely, that when you use synchronization of poedit with Crowdin, you get the approved strings from Crowdin, but of course, your personal changes (suggestions) also appear in poedit.

File: projectDocs/translating/crowdin.md
Comment (CyrilleB79) [inline projectDocs/translating/crowdin.md:41]: More specifically, you need to manually upload the file on Crowdin's web interface and check an auto-approve checkbox when doing this. But if I am a proof reader and unique translator of a language, I cannot just translate in poedit and hope that synchronization will allow to submit my work with auto-approval unfortunately.
Or is there a better way to do things?

File: projectDocs/translating/crowdin.md
Comment (CyrilleB79) [inline projectDocs/translating/crowdin.md:65]:
```suggestion
If you are unsure of meaning of the original interface message, consult automatic comments (also called translator comments) pressing `control+shift+a`.
```

File: projectDocs/translating/crowdin.md
Comment (CyrilleB79) [inline projectDocs/translating/crowdin.md:135]: Wouldn't "with" be better here?

```suggestion
Translation strings can be grouped with a tag to differentiate contexts.
```

File: projectDocs/translating/crowdin.md
Comment (CyrilleB79) [inline projectDocs/translating/crowdin.md:139]: Let's use a more illustrative example:

```suggestion
The string `none` is defined three times:
* without any context to report a lack of background pattern in Microsoft Excel
* with the context tag `symbolLevel`
* with the context tag `espeakVarient`
```

File: projectDocs/translating/crowdin.md
Comment (seanbudd) [reply projectDocs/translating/crowdin.md:135]: I think they are equivalent, no?

Comment (seanbudd) [reply projectDocs/translating/crowdin.md:135]: @coderabbitai which grammar is more correct here? "grouped by" or "grouped with"?

Comment (seanbudd) [reply projectDocs/translating/crowdin.md:135]: interesting, learn some new thing every day huh

File: projectDocs/translating/crowdin.md
Comment (seanbudd) [reply projectDocs/translating/crowdin.md:40]:
```suggestion
PoEdit does not support viewing unapproved strings from other translators.
```

File: projectDocs/translating/crowdin.md
Comment (seanbudd) [reply projectDocs/translating/crowdin.md:41]:
```suggestion
When manually uploading to Crowdin from PoEdit, proofreaders are able to auto-approve all submitted strings.
```

File: projectDocs/translating/crowdin.md
Comment (seanbudd) [inline projectDocs/translating/crowdin.md:136]:
```suggestion
```

File: projectDocs/translating/crowdin.md
Comment (CyrilleB79) [reply projectDocs/translating/crowdin.md:135]: The Code Rabbit AI wrote:
> @seanbudd In the context provided, "grouped by a tag" is more grammatically appropriate. This phrasing clearly indicates that the tag is the criterion used for grouping, which aligns well with the intention of differentiating translation strings based on specific contexts. "Grouped with a tag" could imply that the tag is merely accompanying the strings rather than serving as the basis for grouping.

@seanbudd, the rabbit is correct in explaining both sense but cannot guess what you wanted to write. Maybe it was neither clear to me what you wanted to write.

But "grouped with tag" is more representative of my personal experience. That is, the tag appears along with the string to translate to provide more context (as the translators comment also does).

I am not sure at all that all the strings with context "addonStore" are following each other in the list of strings to translate, making a group of strings related one with another (i.e. "grouped by tage"): at least, it is not the case when untranslated strings are grouped together, either in poedit or on Crowdin's web interface.

It's not a matter of grammar correctness but of meaning.

**Comment by CyrilleB79** [comment]
@seanbudd thanks, you have taken into consideration my comments.

However, I am quite concerned by the way this documentation is written. There seems to be actually 3 ways to translate:
1. Only on Crowdin's web interface, either with only one proof reader translator approving directly their translations or with many translators making suggestions and a proof reading approving them in a second step.
2. With only one proof reader translator translating on poedit without cloud synchronization and uploading manually the file checking the auto-approve checkbox
3. With one ore more translators translating on poedit synchronized with Crowdin's cloud and one proof reader approving the translations on Crowdin.

And the documentation seems to mention parts of the process for one way or the other but does not make a clear distinction between these possible processes, i.e. steps from 2 distinctive processes seems a bit mixed together. The consequence is that a user will have difficulty to identify a step-by-step process to follow.

Note however that I may not master all the possible processes, taking into account that I personally do not feel Crowdin very easy cand comfortable to use. That's why I had asked to other people on the translators mailing list to review this PR. Unfortunately no one has shown up.

Maybe @ruifontes, @Adriani90, @zstanecic would you mind review this PR?

File: projectDocs/translating/crowdin.md
Comment (seanbudd) [reply projectDocs/translating/crowdin.md:135]: I think you are interpreting grouping as being in the same physical group in all list views, as opposed to contextual grouping. These tags are logically grouped based on context, for example, you can search by tag in PoEdit and Crowdin.

Comment (CyrilleB79) [reply projectDocs/translating/crowdin.md:135]: Yes, I was rather interpreting "grouping" as physical grouping rather than contextual grouping.

Anyway, my personal impression is that there is no specific grouping, i.e. "addonStore" strings are not gathered logically. I feel more that the context is an additional information / restriction indicating in which case this translation applies. To summarize, I would not use  "grouping" at all, but rather speak of tags and context only.

Though, it's really a matter of personal and subjective feeling. Feel free to ignore my comments on this topic, the more if no one expresses the same feeling!

**Comment by seanbudd** [comment]
@CyrilleB79 - I'll add a rough outline at the start of the documented covering an overview of these main workflow processes. I think this will help clarify the rest of the instructions

**Comment by seanbudd** [comment]
@CyrilleB79 - I've added a new section

File: projectDocs/translating/crowdin.md
Comment (CyrilleB79) [inline projectDocs/translating/crowdin.md:19]: Fix link:
```suggestion
Alternatively, you can use the [Crowdin web interface](https://support.crowdin.com/online-editor/) directly.
```

**Comment by CyrilleB79** [comment]
@seanbudd I am unsure the document is complete, but I have no other feedback for now. I would have appreciated that other translators comment here but this has not been the case. If no other feedback is received, I'd recommend to merge as is and to update the document in another PR in case we get more feedback in the future. I'll try a last call on the translators mailiing list since the strings for 2024.3 have just been made available for translation.

Also, have you tried yourself all what is written in the document? If not, it would be useful given the lack of received feedback.

**Comment by CyrilleB79** [comment]
On the translators mailing list, someone [offered](https://groups.io/g/nvda-translations/message/4144) to contribute with a tutorial for Crowdin-only usage.
See [this thread](https://groups.io/g/nvda-translations/topic/106737387?p=Created,,,100,1,0,0).
@seanbudd would you accept it?

**Comment by seanbudd** [comment]
@CyrilleB79 - thanks for the feedback. I've responded to the mailing list. And yes, to confirm, I've gone through all the processes with NVDA up to the point of committing any changes.

**Review by michaelDCurran — APPROVED** [review]

**Pull request merged**
